### PR TITLE
CQL Interpolator: Proof of concept

### LINKED ISSUE #45: Spike - CQL Interpolator
Research and discussion about the CQL Interpolator feature

### PR DESCRIPTION
**Pull request by fedefernandez**
### Proof of Concept - DO NOT MERGE

Fixes #45 

This PR shows a proof of concept about how the interpolator could be implemented using [propensive/contextual](https://github.com/propensive/contextual).

It provides a "runtime interpolator" that doesn't make validations at compile time and some utility methods for creating a "compile time interpolator" in the client.

Because we're validating at compile time we need to provide the schema validator at interpolator definition level.

### CODE REVIEW

**Review by juanpedromoreno — APPROVED** [review]
Looks good, I don't know how contextual works, but the code looks good. Also, proof of concept looks nice. Good job!

**Comment by fedefernandez** [comment]
@raulraja do you like the approach? should we move this forward?

File: test/src/main/scala/TestCompileInterpolator.scala [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+object TestCompileInterpolator extends App {
+
+  import freestyle.cassandra.query.interpolator.MyInterpolator._
+
+  import scala.concurrent.ExecutionContext.Implicits.global
+  import cats.instances.future._
+
+  val id = 1
+
+  val r: Future[Statement] = cql"SELECT id, name FROM test.users WHERE id = $id;"
```
Comment (raulraja) [inline test/src/main/scala/TestCompileInterpolator.scala:31]: How would we `Statement => User`?

**Review by raulraja — APPROVED** [review]
what would the compile time version look like from a user point of view?

**Comment by fedefernandez** [comment]
For the compile time version, the user should write an interpolator like [`MyInterpolator`](https://github.com/frees-io/freestyle-cassandra/pull/65/files#diff-7847b5f7a7255ef6eceb430e9b0890d4R23). We could implement a macro for it.

File: test/src/main/scala/TestCompileInterpolator.scala (hunk 1 above)
Comment (fedefernandez) [reply test/src/main/scala/TestCompileInterpolator.scala:31]: This just an example, ideally the interpolator will create a `BoundedStatement` and with the `LowLevelAPI` (or another intermediate algebra) return a list of users.

We need to define how we're going to do the CRUD operations at a low level (#57, #61, #62, #63, #58) and then create this intermediate step. In the meantime, I think it's a good approach just returning a `Statement` value.

**Pull request closed without merging**